### ISSUE TITLE
Use `libtor` to bundle tor directly into the CLI

### ISSUE THREAD

**Issue by binarybaron** [comit-network/xmr-btc-swap#1455]
We could use this library https://github.com/MagicalBitcoin/libtor to bundle tor

**Comment by delta1** [comment]
i have done this before and it doesn’t build natively on windows. still, we can put it behind a feature flag and use that for linux/macos binaries

**Comment by delta1** [comment]
for reference: https://github.com/tari-project/tari/pull/3641#issuecomment-1006557338

**Comment by binarybaron** [comment]
> it doesn’t build natively on windows

that's a bummer :(

**Comment by binarybaron** [comment]
> i have done this before on a different project and it doesn’t build natively on windows. still, we can put it behind a feature flag and use that for linux/macos binaries

But we can build it for windows on linux

**Comment by delta1** [comment]
true, it just means devs on windows can’t build without msys2/mingw

**Comment by sneurlax** [comment]
> true, it just means devs on windows can’t build without msys2/mingw

have had to do this often, unfortunate tho it is
